cdp_scrapers/legistar_utils.py

```diff
@@ -1187,43 +1190,47 @@ def get_content_uris(self, legistar_ev: Dict) -> List[ContentURIs]:
 
     def inject_known_person(self, person: Person) -> Person:
         """
-        Inject information from self.known_persons
-        if person exists in the dictionary for long-term static data
+        Inject information if person exists in known_static_data.persons
 
         Parameters
         ----------
         person: Person
-            Person into which to inject data from known_persons
+            Person into which to inject data from known_static_data
 
         Returns
         -------
         Person
-            Input person updated with more information
-            or as-is if person not found in known_persons
+            Input person updated with information from known_static_data,
+            and seat.roles sanitized.
+
+        See Also
+        --------
+        scraper_utils.sanitize_roles()
         """
         try:
-            known_person = self.known_persons[person.name]
-        except (TypeError, KeyError):
+            known_person = self.known_static_data.persons[person.name]
+        except (AttributeError, KeyError):
             return person
 
         for attr in person.__dataclass_fields__.keys():
             static_info = getattr(known_person, attr)
             if static_info is not None:
-                # have long-term information provided in "static*.json"; just use it
+                # have long-term information provided in "static*.json"
                 setattr(person, attr, static_info)
-            else:
-                # use dynamically obtained information, which may be None
-                setattr(person, attr, getattr(person, attr))
 
         # now that we have seat from static hard-coded data
         # we can bring in seat.roles (OfficeRecords from Legistar API)
-        if person.seat and not person.seat.roles:
-            person.seat.roles = self.get_roles(
-                get_legistar_person(
-                    client=self.client_name,
-                    person_id=person.external_source_id,
-                    use_cache=True,
-                )[LEGISTAR_PERSON_ROLES]
+        if person.seat is not None:
+            person.seat.roles = sanitize_roles(
+                person_name=person.name,
+                roles=self.get_roles(
+                    legistar_office_records=get_legistar_person(
+                        client=self.client_name,
+                        person_id=person.external_source_id,
+                        use_cache=True,
+                    )[LEGISTAR_PERSON_ROLES]
+                ),
+                known_static_data=self.known_static_data,
```

Comment:
Very much like your ideas for deciding what `OfficeRecords` to actually keep. Thank you.

I will just make one correction in your comment.

> the roles returned by sanitize_roles will have person.seat as their seat

The answer is actually no, because in the ingestion models, `Seat` is a [member field of `Person`](https://github.com/CouncilDataProject/cdp-backend/blob/f056a4a2825e1584eb2085000bb2d053a91be6c7/cdp_backend/pipeline/ingestion_models.py#L50), not `Role`.

Reply:
I meant to say the roles returned by sanitize_roles will eventually make it into the DB, and their seat will be determined by person.seat (of static file)

Reply:
Correct.